robottelo/cli/hammer.py

```diff
@@ -0,0 +1,163 @@
+"""Helpers to interact with hammer command line utility."""
+import re
+
+from itertools import izip
+
+
+def parse_csv(output):
+    """Parse CSV output from Hammer CLI and convert it to python dictionary."""
+    result = []
+    # Generate the key names, spaces will be converted to dashes "-"
+    keys = [
+        header.replace(' ', '-').lower() for header in output.pop(0).split(',')
+    ]
+
+    # For each entry, split the values and create a dict mapping each key with
+    # each value
+    for line in output:
+        if len(line) > 0:
+            result.append(dict(izip(keys, line.split(','))))
+
+    return result
+
+
+def parse_help(output):
+    """Parse the help output from a hammer command and return a dictinary
+    mapping the subcommands and options accepted by that command.
+
+    """
+    # Parsing states
+    state = 0
+    subcommands_section_state = 1
+    options_section_state = 2
+
+    contents = {
+        'subcommands': [],
+        'options': [],
+    }
+    option_regex = re.compile(
+        r'^ (-(?P<shortname>\w), )?(--(?P<name>[\w-]+))?( (?P<value>\w+))?\s+'
+        '(?P<help>.*)$'
+    )
+    subcommand_regex = re.compile(
+        r'^ (?P<name>[\w-]+)?\s+(?P<description>.*)$'
+    )
+
+    for line in output:
+        if len(line.strip()) == 0:
+            continue
+        if line.startswith('Subcommands:'):
+            state = subcommands_section_state
+            continue
+        if line.startswith('Options:'):
+            state = options_section_state
+            continue
+
+        if state == subcommands_section_state:
+            match = subcommand_regex.search(line)
+            if match is not None:
+                if match.group('name') is None:
+                    contents['subcommands'][-1]['description'] += (
+                        ' {0}'.format(match.group('description'))
+                    )
+                else:
+                    contents['subcommands'].append({
+                        u'name': match.group('name'),
+                        u'description': match.group('description'),
+                    })
+        if state == options_section_state:
+            match = option_regex.search(line)
+            if match is not None:
+                if match.group('name') is None:
+                    contents['options'][-1]['help'] += (
+                        ' {0}'.format(match.group('help'))
+                    )
+                else:
+                    contents['options'].append({
+                        u'name': match.group('name'),
+                        u'shortname': match.group('shortname'),
+                        u'value': match.group('value'),
+                        u'help': match.group('help'),
+                    })
+
+    return contents
+
+
+def parse_info(output):
+    """Parse the info output and returns a dict mapping the values."""
+    # info dictionary
+    r = {}
```

Comment:
Just moved this function to later refactor it, but will change this var name for now and will refactor other things later